dotcom-rendering/src/components/AdSlot.web.tsx

```diff
@@ -36,19 +36,29 @@ type SlotNamesWithPageSkin = SlotName | 'pageskin';
 type InlineProps = {
 	position: InlinePosition;
 	index: number;
+	shouldHideReaderRevenue?: never;
 };
 
-type NonInlineProps = {
-	position: Exclude<SlotNamesWithPageSkin, InlinePosition>;
+type RightProps = {
+	position: 'right';
 	index?: never;
+	shouldHideReaderRevenue: boolean;
+};
+
+type RemainingProps = {
+	position: Exclude<SlotNamesWithPageSkin, InlinePosition | 'right'>;
+	index?: never;
+	shouldHideReaderRevenue?: never;
 };
 
 /**
- * This allows us to conditionally require the index property based
- * on position. If `position` is an inline type then we expect the
- * index value. If not, then we explicitly refuse this property
+ * This allows us to conditionally require properties based on position:
+ *
+ * - If `position` is an inline type then we expect the `index` prop.
+ * - If `position` is `right` then we expect the `shouldHideReaderRevenue` prop
+ * - If not, then we explicitly refuse these properties
  */
-type Props = DefaultProps & (InlineProps | NonInlineProps);
+type Props = DefaultProps & (RightProps | InlineProps | RemainingProps);
```

Comment:
This TypeScript change was actually super beneficial to performing the refactor here, as it highlighted the exact 3 places I needed to add the `shouldHideReaderRevenue` prop, whilst leaving all of the other ad slots as they are.